Lib/typing.py

```diff
@@ -1136,8 +1136,18 @@ def __mro_entries__(self, bases):
         if self.__origin__ not in bases:
             res.append(self.__origin__)
         i = bases.index(self)
+        # The goal here is to only add Generic to the MRO if nothing else in the
+        # MRO is already a subclass of Generic; otherwise we risk failure to
+        # linearize a consistent MRO.
         for b in bases[i+1:]:
-            if isinstance(b, _BaseGenericAlias) or issubclass(b, Generic):
+            if isinstance(b, _BaseGenericAlias):
+                break
+            if not isinstance(b, type):
+                meth = getattr(b, "__mro_entries__", None)
+                nb = meth(bases) if meth else None
```

Comment:
`nb` is a confusing variable name here, because `b` is a single item in `__orig_bases__`, whereas `nb` is (hopefully) a tuple of items. I'd prefer a name like `new_bases`

Reply:
Yes, I agree, I originally had `new_bases` here and changed it for line-length reasons, but that was not a good choice, and the line-length reasons will probably go away as I handle some additional test cases anyway.